docs/expansions_and_fields.rst

```diff
@@ -148,6 +162,38 @@ Specify the desired fields in a comma-separated list without spaces between
 commas and fields. These specified user fields will display directly in the
 user data objects.
 
+.. _dm_event_fields_parameter:
+
+``dm_event_fields``
+---------------
+
+For methods that return DirectMessageEvents, this `fields`_ parameter enables
+you to select which specific `Direct Message event fields`_ will deliver in
+each returned DirectMessageEvent. Specify the desired fields in a comma-
+separated list without spaces between commas and fields. When requesting DM
+events, there are three default event object attributes, or fields, included:
+``id``, ``event_type``, and ``text``. To receive additional event fields, use
+the fields parameter ``dm_event_fields`` to select others. Other available
+event fields include the following: ``dm_conversation_id``, ``created_at``,
+``sender_id``, ``attachments``, ``participant_ids``, and ``referenced_tweets``.
+
+Several of these fields provide the IDs of other Twitter objects related to the
+Direct Message event:
+
+* ``sender_id`` - The ID of the account that sent the message, or who invited
+a participant to a group conversation
+* ``partricipants_ids`` - An array of account IDs. For ParticipantsJoin and
+ParticipantsLeave events this array will contain a single ID of the account
+that created the event
+* ``attachments`` - Provides media IDs for content that has been uploaded to
+Twitter by the sender
+* ``referenced_tweets`` - If a Tweet URL is found in the text field, the ID of
+that Tweet is included in the response
+
+The ``sender_id``, ``participant_ids``, ``referenced_tweets.id``, and
+``attachments.media_keys`` `expansions`_ are available to expand on these Twitter
+object IDs.
+
```

Comment:
Similarly, the sections like this for other object types are taken from the description for the respective fields parameter for endpoints that return each type of object, e.g. https://developer.twitter.com/en/docs/twitter-api/direct-messages/lookup/api-reference/get-dm_conversations-with-participant_id-dm_events for DM events.

However, you can see that the documentation for `dm_event.fields` is different, so this section is unnecessary. Again, I have plans to improve this documentation as a whole to include documentation from https://developer.twitter.com/en/docs/twitter-api/expansions, so for now, for consistency, this section shouldn't be here.

Reply:
I see. I adapted the description here https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/dm-events to make this section in our doc. If it is not necessary then l'll remove line 165-195.

Many thanks!

Reply:
I'll remove this section, but if we want some completeness before the next big doc improvement, we can put the following code back later just in case people thought the doc is not up-to-date:

```rst
.. _dm_event_fields_parameter:

``dm_event_fields``
---------------

Extra fields to include in the event payload. `id`, `text` and `event_type` are
returned by default.

.. _Direct Message event fields: https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/dm-events
```

Reply:
> Oh I got it. You meant the adapted intro from https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/dm-events is not necessary, and, for now, I should use the _description_ for `dm_event.fields`. However, there is only one endpoint using `dm_event.fields` so I don't need to complete this documentation?

Correct, I've added that documentation to `DirectMessageEvent` itself now, with https://github.com/tweepy/tweepy/commit/c689548cb1b4b9c2f69249d67534a735b931b9b5.

Reply:
> I'll remove this section, but if we want some completeness in this doc page before the next big doc improvement (and also for the consistency in Client methods' docstrings), we can put the following code back later just in case people thought the doc is not up-to-date:
> 
> ```
> .. _dm_event_fields_parameter:
> 
> ``dm_event_fields``
> ---------------
> 
> Extra fields to include in the event payload. `id`, `text` and `event_type` are
> returned by default.
> 
> .. _Direct Message event fields: https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/dm-events
> ```

For the sake of completeness, I've added a section for this now, with https://github.com/tweepy/tweepy/commit/4b0fa90e91eb2b67dfd33f0d27b148e95ea05f65.
I don't think it's necessary to replace or add it to the documentation for the `Client` and `AsyncClient` methods that have this parameter yet though.